ddtrace/contrib/aiokafka/patch.py

```diff
@@ -0,0 +1,258 @@
+        dd_headers[header[0]] = header[1].decode("utf-8")
+    return dd_headers
+
+
+def create_get_span_tags(instance):
+    tags = {
+        COMPONENT: config.aiokafka.integration_name,
+        SPAN_KIND: SpanKind.CONSUMER,
+        HOST_LIST: bootstrap_servers(instance),
+        MESSAGING_SYSTEM: SERVICE,
+        GROUP_ID: instance._group_id,
+    }
+    return tags
+
+
+async def traced_getone(func, instance, args, kwargs):
+    pin = Pin.get_from(instance)
+    if not pin or not pin.enabled():
+        return await func(*args, **kwargs)
+
+    err = None
+    result = None
+    child_of = None
+    try:
+        result = await func(*args, **kwargs)
+        if config.aiokafka.distributed_tracing_enabled and result.headers:
+            child_of = HTTPPropagator.extract(convert_headers_for_distributed_tracing(result.headers))
+    except Exception as e:
+        err = e
+        raise err
+    finally:
+        with core.context_with_data(
+            "aiokafka.getone",
+            call_trace=False,
+            span_name=schematize_messaging_operation(CONSUME, provider="kafka", direction=SpanDirection.INBOUND),
+            span_type=SpanTypes.WORKER,
+            service=trace_utils.ext_service(pin, config.aiokafka),
+            call_key="instrumented_getone",
+            distributed_context=child_of,
+            tags=create_get_span_tags(instance),
+            pin=pin,
+        ) as ctx, ctx[ctx["call_key"]] as span:
+            core.dispatch("aiokafka.getone.message", (ctx, result, err))
+            core.dispatch("aiokafka.get.completed", (instance, result, span))
+        return result
+
+
+async def traced_getmany(func, instance, args, kwargs):
+    pin = Pin.get_from(instance)
+    if not pin or not pin.enabled():
+        return await func(*args, **kwargs)
+
+    err = None
+    result = None
+    try:
+        result = await func(*args, **kwargs)
+    except Exception as e:
+        err = e
+        raise err
+    finally:
+        for tp, records in result.items():
+            for record in records:
+                child_of = None
+                if config.aiokafka.distributed_tracing_enabled and record.headers:
+                    child_of = HTTPPropagator.extract(convert_headers_for_distributed_tracing(record.headers))
+                with core.context_with_data(
+                    "aiokafka.getmany",
+                    call_trace=False,
+                    span_name=schematize_messaging_operation(
+                        CONSUME, provider="kafka", direction=SpanDirection.INBOUND
+                    ),
+                    span_type=SpanTypes.WORKER,
+                    service=trace_utils.ext_service(pin, config.aiokafka),
+                    call_key="instrumented_getmany",
+                    distributed_context=child_of,
+                    tags=create_get_span_tags(instance),
+                    pin=pin,
+                ) as ctx, ctx[ctx["call_key"]] as span:
```

Comment:
Unfortunately, our current trace model does not support many-to-many parent-child propagation. Generally, for batch message operations we have taken the following approach:
- If the batch operation calls `aiokafka.getone` n times for n records, then we will want to create a span for each of these calls. Many times this is not the case, and the batch operation will actually be 1 operation, returning N records. For each of these n spans, we set a span link to the original producer.  
- In the case of 1 operation returning N records, we should create 1 span for the batch operation, similar to `aiokafka.getone`. Again, we will not support parent-child propagation. Instead, we will use span links and set n links on the single span, each to their respective producer. 

Again, thank you for the contribution, and please don't hesitate to reach out for help. Also, if you would like, I could make direct changes to your branch to speed up the process. 

(FYI, span links are very new to ddtrace, and there may be limited examples to follow. We were planning to begin using them for auto-instrumentation this quarter)

Reply:
@wconti27 feel free to many any changes as you see fit. Especially for the span link part.